modules/apps/29-fee/ibc_module.go

```diff
@@ -200,7 +200,7 @@ func (im IBCModule) OnRecvPacket(
 
 	// incase of async aknowledgement (ack == nil) store the ForwardRelayer address for use later
 	if ack == nil && found {
-		im.keeper.SetForwardRelayerAddress(ctx, channeltypes.NewPacketId(packet.GetSourceChannel(), packet.GetSourcePort(), packet.GetSequence()), forwardRelayer)
+		im.keeper.SetForwardRelayerAddress(ctx, channeltypes.NewPacketId(packet.GetDestChannel(), packet.GetDestPort(), packet.GetSequence()), forwardRelayer)
```

Comment:
Is there a test for this so we can catch the issue if it occurs again?

Reply:
It's a good point.

Reply:
failing here -> https://github.com/cosmos/ibc-go/pull/961/files#diff-79b2cea153b51838e9cb02a38178bc5bc0eb313f75d74f8988774ad0d492e957R541

Reply:
Alright, I have instead moved the check for WriteAcknowledgement to the tests in `relay_test.go` instead as it was easier and made more sense. We have full coverage now so if either of these lines are accidentally set to `SourceChannel` instead of `DestinationChannel` the tests will fail.